File: software/examples/java/ExampleSimple.java
import com.tinkerforge.BrickletCurrent12;
import com.tinkerforge.IPConnection;

public class ExampleSimple {
	private static final String host = new String("localhost");
	private static final int port = 4223;
	private static final String UID = new String("ABC"); // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the commnents below
	public static void main(String args[]) throws Exception {
		// Create connection to brickd
		IPConnection ipcon = new IPConnection(host, port); // Can throw IOException

		BrickletCurrent12 c12 = new BrickletCurrent12(UID); // Create device object

		// Add device to ip connection
		ipcon.addDevice(c12); // Can throw IPConnection.TimeoutException
		// Don't use device before it is added to a connection
		

		// Get current current (unit is mA)
		short current = c12.getCurrent(); // Can throw IPConnection.TimeoutException

		System.out.println("Current: " + current/1000.0 + " A");
		
		System.out.println("Press ctrl+c to exit");
		ipcon.joinThread();
	}
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/current12_bricklet"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Current12 Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	c, _ := current12_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current current.
	current, _ := c.GetCurrent()
	fmt.Printf("Current: %f A\n", float64(current)/1000.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
